py/gc.c

```diff
@@ -258,10 +258,6 @@ STATIC void gc_sweep(void) {
             case AT_HEAD:
 #if MICROPY_ENABLE_FINALISER
                 if (FTB_GET(block)) {
-                    #if MICROPY_PY_THREAD
-                    // TODO need to think about reentrancy with finaliser code
-                    assert(!"finaliser with threading not implemented");
-                    #endif
```

Comment:
OK... I'll leave that one to you. So far we don't allow any Python code to run within interrupts context, so we should be fine.